File: C++/can-make-arithmetic-progression-from-sequence.cpp
class Solution {
public:
    bool canMakeArithmeticProgression(vector<int>& arr) {
        sort(arr.begin(), arr.end());
        int delta = arr[1] - arr[0];

        int sz = arr.size();

        for (int i = 1; i < sz; i++) {
            if (arr[i] - arr[i-1] != delta) {
                return false;
            }
        }

        return true;
    }
};

File: TypeScript/can-make-arithmetic-progression-from-sequence.ts
function canMakeArithmeticProgression(arr: number[]): boolean {
    arr.sort((a, b) => a - b)
    const delta = arr[0] - arr[1]
    for (let i = 2; i < arr.length; i++) {
        if (arr[i-1] - arr[i] !== delta) {
            return false
        }
    }
    return true
};